### DISCUSSION TITLE
Adding constraint in the model output parameter

### DISCUSSION THREAD

**Discussion by Pranay-Nagrani** [lululxvi/deepxde#853] [General]
Hi Dr. Lu,

I am trying to solve an inverse problem using deepxde in which there are two model outputs y1 and y2 and two inputs x1 and x2. I am wondering if there is a trick to constraint y2 to be strictly between 0 and 1, i.e. 0<=y2<=1. 

Similarly, is there a way to ensure the unknown model parameter which is being learned during the training is strictly positive?

Thanks for your help.

Pranay

**Comment by forxltk** [comment]
The basic idea is to use surrogate function with upper and lower bound. See https://github.com/lu-group/sbinn

**Comment by Pranay-Nagrani** [comment]
Thank you for the paper. I believe it shows how to implement a search range in unknown parameters. Is there anyway to limit the output to be between 0 and 1?

**Comment by lululxvi** [comment]
Use `output_transform` in a similar way, see https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/lotka.volterra.html?highlight=output_transform#lotka-volterra-equation

**Comment by Pranay-Nagrani** [comment]
Great, thank you very much.

**Reply by tjboise** [reply]
can you share how do you solve this problem with the ```output_transform``` ? appreciate
